qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,438 @@
+                x, amp=amp1, freq=freq1, phase=-2 * np.pi * freq1 * beta, baseline=base1
+            ),
+            plot_color="green",
+            name="series-1",
+            filter_kwargs={"series": 1},
+            plot_symbol="^",
+        ),
+        SeriesDef(
+            fit_func=lambda x, amp0, amp1, amp2, freq0, freq1, freq2, beta, base0, base1, base2: cos(
+                x, amp=amp2, freq=freq2, phase=-2 * np.pi * freq2 * beta, baseline=base2
+            ),
+            plot_color="red",
+            name="series-2",
+            filter_kwargs={"series": 2},
+            plot_symbol="v",
+        ),
+    ]
+
+    @classmethod
+    def _default_options(cls):
+        """Return the default analysis options.
+
+        See :meth:`~qiskit_experiment.analysis.CurveAnalysis._default_options` for
+        descriptions of analysis options.
+        """
+        default_options = super()._default_options()
+        default_options.p0 = {
+            "amp0": None,
+            "amp1": None,
+            "amp2": None,
+            "freq0": None,
+            "freq1": None,
+            "freq2": None,
+            "beta": None,
+            "base0": None,
+            "base1": None,
+            "base2": None,
+        }
+        default_options.bounds = {
+            "amp0": None,
+            "amp1": None,
+            "amp2": None,
+            "freq0": None,
+            "freq1": None,
+            "freq2": None,
+            "beta": None,
+            "base0": None,
+            "base1": None,
+            "base2": None,
+        }
+        default_options.fit_reports = {"beta": "beta"}
+        default_options.xlabel = "Beta"
+        default_options.ylabel = "Signal (arb. units)"
+
+        return default_options
+
+    def _setup_fitting(self, **options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+        """Compute the initial guesses."""
+        user_p0 = self._get_option("p0")
+        user_bounds = self._get_option("bounds")
+
+        # Use a fast Fourier transform to guess the frequency.
+        x_data = self._data("series-0").x
+        delta_beta = x_data[1] - x_data[0]
+
+        min_beta, max_beta = min(x_data), max(x_data)
+
+        freq_guess = []
+        amp_guesses = []
+        base_guesses = []
+        for series in ["series-0", "series-1", "series-2"]:
+            y_data = self._data(series).y
+            fft = np.abs(np.fft.fft(y_data - np.average(y_data)))
+            freqs = np.linspace(0.0, 1.0 / (2.0 * delta_beta), len(fft))
+            freq_guess.append(freqs[np.argmax(fft[0 : len(fft) // 2])])
+
+            b_guess = np.average(y_data)
+            amp_guesses.append(np.max(y_data) - np.min(y_data) - b_guess)
+            base_guesses.append(b_guess)
+
+        if user_p0.get("beta", None) is not None:
+            p_guesses = [user_p0["beta"]]
+        else:
+            p_guesses = np.linspace(min_beta, max_beta, 20)
```

Comment:
This should be fine. We have been using the beta for which the series have the minimum separation.